Add page title to search index

kw-pr opened the pull request:
The title is kind of important.
I add the title to the first CMS_HTML element... maybe someone can clean this up.

kw-pr committed:
Add page title to search index

The title is kind of important.
I add the title to the first CMS_HTML element... maybe someone can clean this up.

muratpurc commented:
What is the purpose of this pr, is it a bug or an enhancement, like to just have the title be indexed?

The change in the code retrieves the first matching entry from the art_lang-table, even if the current indexed language is not the same. Why did you comented out the `AND idlang = ...` part?

kw-pr commented:
I think the title of the page contains the most important key words. That is also why for example google does rank those higher. It is probably a matter of opinion if it is called a bug or an enhancement. I guess a bit of both.

Why did I comment it out? Excellent question! Not sure... I only have one language, it worked for me this way, but I guess I already know that I should probably add something about the language. This needs some more work...

muratpurc commented:
@kw-pr I like the change, but we should always add changes to the CONTENIDO source code that work for all CONTENIDO users. Also please stick to the formats, if everyone add their own style of programming, we will end up with ugly source code. CONTENIDO should be compatible with older PHP versions as far as possible, we should not use any features of new PHP versions.

Please adjust your PR as follows:
1. Remove the return type `:void` of the function `addTittle()`, this is not compatible with PHP 7.0, CONTENIDO should still be able to run under PHP 7.0, even if most providers are moving away from it. In addition, we have not yet used any return types in CONTENIDO.
2. Don't put opening curly braces on new lines, see the rest of the file's source code.
3. You can delete the following lines:
```
         global $cfg;
         $oDB = cRegistry::getDb();
```
You don't need them because these variables already exist as properties of the parent class, you can access them with `$this->cfg` and `$this->db`.
4. Only two fields are needed from the table `art_lang`, so you don't have to load all fields with `*`
5. The language `AND idlang = " . cSecurity::toInteger($this->lang);` should also be added to the SQL statement so that it works properly in multilingual CONTENIDO installations

muratpurc committed:
#271 PR - Add page title to search index.

Reworked pull request code, added cArray::getFirstKey() instead of using array_key_first() which is available PHP >= 7.3.0.
